easybuild/framework/easyblock.py

```diff
@@ -2103,7 +2106,9 @@ def install_extensions_sequential(self, install=True):
                 # don't reload modules for toolchain, there is no need since they will be loaded already;
                 # the (fake) module for the parent software gets loaded before installing extensions
                 ext.toolchain.prepare(onlymod=self.cfg['onlytcmod'], silent=True, loadmod=False,
-                                      rpath_filter_dirs=self.rpath_filter_dirs)
+                                      rpath_filter_dirs=self.rpath_filter_dirs,
+                                      rpath_include_dirs=self.rpath_include_dirs,
```

Comment:
The omission of this looks like a bug to me, but I can't be 100%

Reply:
It probably doesn't actually matter as the wrappers will already have been prepared by the time this is called by an extension.

Reply:
(see https://github.com/easybuilders/easybuild-framework/pull/4596/files#diff-c1bad9950fee9f74b59e255119ea04ce41e9d9ce8cfa9fa94e831c8d314d106dR1041)

Reply:
It has no impact for 2nd extension, where we'll pick up on wrapper created for first extension.

But the RPATH wrapper that was created for the "parent" will no longer be found (because the build environment is reset before build environment for first extension is set up), and so the installation of extensions will be done with an incorrect RPATH wrapper script in which `rpath_include_dirs` is not picked up.

That's bad since `$ORIGIN/../lib` & co are specified that way (see how `self.rpath_wrappers_dir` is defined in `EasyBlock.prepare_step`).

Reply:
Hold on, the resetting of build environment before starting installation of extensions is not correct I think, I'm going through things in detail now.

Reply:
The `rpath_include_dirs` mechanism was implemented in #2358, which shows that it's only picked up by the "parent" `toolchain.prepare`, the call to `toolchain.prepare` for extensions was overlooked. That confirms to me this is a genuine bug.

I'm still a bit puzzled why this doesn't cause real trouble though, or how it has flown under the radar...